### PR TITLE
Move OpenAPI lint check to unified CI

### LINKED ISSUE #24267: Integrate OpenAPI lint check into main CI
## Description

The OpenAPI lint check introduced with https://github.com/camunda/camunda/pull/23806 is a separate check that is not required to pass for PRs.

## Acceptance criteria

The OpenAPI lint check is part of the [Unified CI](https://github.com/camunda/camunda/wiki/CI-&-Automation#implementation).

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

Moves the OpenAPI lint check to the unified CI and makes it required.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [x] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #24267

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -546,6 +547,31 @@ jobs:
           secret_vault_secretId: ${{ secrets.VAULT_SECRET_ID }}
           secret_vault_address: ${{ secrets.VAULT_ADDR }}
           secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
+  
+  openapi-lint:
+    name: C8 REST OpenAPI linting
+    timeout-minutes: 10
```
Comment (cmur2) [inline .github/workflows/ci.yml:553]: You could consider lowering this to reflect more accurately how much time is reasonably needed (few minutes I guess).

**Review by cmur2 — APPROVED** [review]

**Comment by cmur2** [comment]
@tmetzke can you please also delete the other (redundant) standalone workflow, either in this PR or separately?

**Comment by tmetzke** [comment]
> @tmetzke can you please also delete the other (redundant) standalone workflow, either in this PR or separately?

Good catch, totally forgot about it 😅

**Comment by tmetzke** [comment]
Linting is invoked and works as expected, see https://github.com/camunda/camunda/actions/runs/11685379830/job/32538801254?pr=24367 👍

**Pull request merged**